src/main/java/com/techsophy/tsf/account/controller/impl/UserFormDataControllerImpl.java

```diff
@@ -39,8 +39,26 @@ public ApiResponse<AuditableData> getUserDetailsOfLoggedInUser() {
     }
     @Override
     public ApiResponse<UserFormDataSchema> updateUserDetailsOfLoggedInUser(UserFormDataSchema userFormDataSchema) {
-        return new ApiResponse<>(userFormDataService.saveUserFormData(userFormDataSchema),true,"updated Successfully");
+        try {
+            String userId = (String) userDetails.getUserDetails().get(0).get(ID);
+            if (userFormDataSchema.getUserId() != null && userFormDataSchema.getUserId().equalsIgnoreCase(userId)) {
+                return new ApiResponse<>(userFormDataService.saveUserFormData(userFormDataSchema), true, "updated Successfully");
```

Comment:
add these string as constants and better to retrieve from global message source so that they can be internationalized.